sound/soc/sof/control.c

```diff
@@ -11,8 +11,35 @@
 /* Mixer Controls */
 
 #include <linux/pm_runtime.h>
+#include <linux/leds.h>
 #include "sof-priv.h"
 
+static void update_acpi_mute_led(struct snd_sof_control *scontrol,
+				 struct snd_kcontrol *kcontrol,
+				 struct snd_ctl_elem_value *ucontrol)
+{
+	unsigned int temp = 0;
+	unsigned int mask;
+
+	mask = 1U << snd_ctl_get_ioffidx(kcontrol, &ucontrol->id);
+
+	if (ucontrol->value.integer.value[0] ||
+	    (scontrol->num_channels == 2 && ucontrol->value.integer.value[1]))
+		temp |= mask;
```

Comment:
@juimonen what I meant was something like this. I'm assuming that num_channels can be > 2 here. If thats not the case then it doesnt matter
temp &= ~mask;
for (i = 0; i < scontrol->num_channels; i++)
      if (ucontrol->value.integer.value[i] {
            temp |=mask;
            break;
     }

Reply:
ok, let my try this tomorrow in a real device.

Reply:
@ranj063 updated to your suggestion, seems to work nicely, thanks.